### PR TITLE
fix: loopMenuItemRecursively error children.forEach is not a function

### PR DESCRIPTION
**Pull request by one-pupil**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/vc-menu/util.js [hunk 1]
```diff
@@ -29,10 +29,11 @@ export function getMenuIdFromSubMenuEventKey(eventKey) {
 // }
 
 export function loopMenuItemRecursively(children, keys, ret) {
-  if (!children || ret.find) {
+  // fix: children.forEach is not a function
+  if (!children || ret.find || !Array.isArray(children)) {
```
Comment (tangjinzhou) [inline components/vc-menu/util.js:33]: 什么场景会出现 children 不是数组？

Comment (one-pupil) [reply components/vc-menu/util.js:33]: 我在处理
image
异常的时候
image
```js
props.menuInfo.children.map(item => {
    if (!item.children) {
      return (
        <Menu.Item key={item.key}>
          {
            () => <span>{item.name}</span>
          }
        </Menu.Item>
      );
    } else {
      return <SubMenu menuInfo={item} key={item.key} />;
    }
  })
```

image

**Comment by tangjinzhou** [comment]
@Amour1688 是否是 jsx 问题
```jsx
<Menu.Item key={item.key}>
          {
            () => <span>{item.name}</span>
          }
        </Menu.Item>
```

**Comment by chad-bin** [comment]
这个是vite1.0 tsx会出现的问题。只需改成`[<span>{}</span>]`就可以。vi te2.0引入了tsx插件应该可以解决这个问题

**Pull request closed without merging**